<a href="https://colab.research.google.com/github/GODRATAN/FlyRank_ML-Intern/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/GODRATAN/FlyRank_ML_Assignment/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Content Refresh Prioritization

1. I chose the Content Refresh Prioritization lane because the goal is to help
2. content editors decide which pages should be reviewed first.
3. The starter dataset contains information such as impressions, CTR,
4. ranking position, and content age that can help identify pages showing
5. signs of decline.
6. This project focuses on supporting editorial decisions by prioritizing
7. pages that may benefit from a content refresh.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Decision, Action, and Cost

Decision:
-Which content pages should be prioritized for refresh?

Who acts?
-Content editors or SEO managers.

Action:
-Review and update the highest-priority pages first.

 Cost of a wrong recommendation:
- If a page is incorrectly prioritized, editors may spend time updating
- content that did not need attention.
- If a declining page is missed, it may continue losing traffic and
- visibility before anyone notices.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import os
import sys
import subprocess
import pandas as pd

REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print(df.shape)

(30000, 44)


In [7]:
# Supporting numbers from the starter dataset

visible = df[df["impressions_90d"] > 0]
corr = visible["search_volume"].corr(visible["impressions_90d"])

print(f"Correlation (search volume vs impressions): {corr:.3f}")

print(f"Total pages in dataset: {len(df)}")

declining = (df["trend_direction"].str.lower() == "down").sum()
print(f"Pages marked as declining: {declining}")

Correlation (search volume vs impressions): 0.001
Total pages in dataset: 30000
Pages marked as declining: 16262


### Observation

The starter dataset contains **30,000** content pages, of which **16,262** are marked as declining. After filtering to pages with impressions greater than zero, the correlation between search volume and impressions is **0.001**, indicating almost no linear relationship. These observations suggest that relying on search volume alone may not be sufficient for prioritizing content refreshes, making it worthwhile to investigate multiple signals during the internship.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Claims

1. This work is intended to support editorial decision-making.
2. The findings are based on the provided starter dataset.
3. The analysis is observational and directional rather than causal.
4. It does not prove how search engines rank content.
5. It does not guarantee that refreshing a page will improve performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.